torchmetrics/detection/map.py

```diff
@@ -343,6 +377,16 @@ def _get_classes(self) -> List:
         return []
 
     def _compute_iou(self, id: int, class_id: int, max_det: int) -> Tensor:
+        if self.iou_type == "segm":
+            return self._compute_iou_impl(id, self.groundtruth_masks, self.detection_masks, class_id, max_det, segm_iou)
+        elif self.iou_type == "bbox":
+            return self._compute_iou_impl(id, self.groundtruth_boxes, self.detection_boxes, class_id, max_det, box_iou)
+        else:
+            raise Exception(f"IOU type {self.iou_type} is not supported")
```

Comment:
the exception case is already handled in `__init__`
```suggestion
        return self._compute_iou_impl(
          id,
          class_id, 
          max_det, 
          segm_iou if self.iou_type == "segm" else box_iou
        )
```